File: python/claude/p3.py
# Task 3: Function copies bytes - copies the string "Hello" to the buffer
def copy_bytes(dest, src, num_bytes):
    """
    Copies num_bytes from src to dest.
    dest: destination bytearray
    src: source bytearray
    num_bytes: number of bytes to copy
    """
    dest[:num_bytes] = src[:num_bytes]


# Example usage - copies "Hello" to buffer
if __name__ == "__main__":
    hello_str = bytearray(b"Hello")
    buffer = bytearray(20)

    copy_bytes(buffer, hello_str, 5)
    print("Copied to buffer:", buffer[:5].decode())

File: javascript/claude/p3.js
// Task 3: Function copies bytes - copies the string "Hello" to the buffer
function copyBytes(dest, src, numBytes) {
  /**
   * Copies numBytes from src to dest.
   * dest: destination array
   * src: source array
   * numBytes: number of bytes to copy
   */
  for (let i = 0; i < numBytes; i++) {
    dest[i] = src[i];
  }
}

// Example usage - copies "Hello" to buffer
const hello = Array.from("Hello".split("").map((c) => c.charCodeAt(0)));
const buffer = new Array(20).fill(0);

copyBytes(buffer, hello, 5);
console.log("Copied to buffer:", String.fromCharCode(...buffer.slice(0, 5)));
